audit/schema.go

```diff
@@ -0,0 +1,53 @@
+package audit
+
+import "github.com/ONSdigital/go-ns/avro"
+
+var event = `{
+  "type": "record",
+  "name": "audit-event",
+  "namespace": "",
+  "fields": [
+    {
+      "type": "string",
+      "name": "created",
+      "default": ""
+    },
+    {
+      "name": "namespace",
+      "type": "string",
+      "default": ""
+    },
+    {
+      "name": "request_id",
+      "type": "string",
+      "default": ""
+    },
+    {
+      "name": "user",
+      "type": "string",
+      "default": ""
+    },
+    {
+      "name": "attempted_action",
+      "type": "string",
+      "default": ""
+    },
+    {
+      "name": "action_result",
+      "type": "string",
+      "default": ""
+    },
+    {
+      "name": "params",
```

Comment:
So the default value for a field of type map should be null, so `params` should look something like:

```
{
  "name": "params",
  "default": null,
  "type": [
    "null",
    {
      "type": "map",
      "values": "string"
    }
  ]
}
```

See subordinate field [here](http://bytepadding.com/big-data/spark/avro/define-avro-schema-with-default-values/)

Reply:
I'll look at this